### DISCUSSION TITLE
APIs do IBGE

### DISCUSSION THREAD

**Discussion by frontendwizard** [BrasilAPI/BrasilAPI#236] [Show and tell]
Em particular as APIs de malha geográfica e de localidades são úteis para cria gráficos e mapas sobre o brasil. A api de nomes também pode ser interessante.

https://servicodados.ibge.gov.br/api/docs

O que vocês acham?

**Comment by pereira299** [comment]
@thefrontendwizard acho bastante interessante as malhas geográficas do IBGE, contudo por enquanto eles só possuem malhas para regiões e estados. Por outro lado, se não me engano, no Wikipedia Images há SVGs dos estados, subdividido em mesorregião, microrregião e municípios.

https://commons.wikimedia.org/wiki/Category:Maps_of_Brazil

**Comment by frontendwizard** [comment]
@pereira299 
a API de malhas do IBGE tem um parametro chamado resolução que define qual o nivel de detalhamento da malha:
```
0. Nenhuma divisão político-administrativa é inserida no interior da malha
1. Inclui na malha as macrorregiões. Válido apenas quando a localidade for BR.
2. Inclui na malha as Unidades da Federação. Válido apenas quando a localidade for BR ou uma macroregião
3. Inclui na malha as mesorregiões. Válido apenas quando a localidade for BR, macroregião ou Unidade da Federação
4. Inclui na malha as microrregiões. Válido apenas quando a localidade for BR, macroregião, Unidade da Federação ou mesorregião
5. Inclui na malha os municípios
```
Além disso também disponibiliza não só svgs mas também geoJSON e topojson, assim como o nivel de qualidade das topologias (nivel de detalhamento das fronteiras) que resulta em svgs e topologias mais leves pra renderizar.

**Comment by frontendwizard** [comment]
Dito isto, a coleção da wikipedia é bem impressionante e vai um pouco além. Só que talvez sirva a diferentes propósitos. Meu maior interesse nas apis do IBGE são os topojson que posso usar para renderizar os mapas criar visualizações usando a geografia do brasil, o que não seria simples nem trivial fazer com SVGs.
